src/components/SearchableDropdown/index.tsx

```diff
@@ -0,0 +1,237 @@
+
+  // Handle onClickOutside
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+    const onClickOutside = (event: MouseEvent) => {
+      const stillFocused =
+        menuButtonRef.current!.contains(event.target as Node) ||
+        listMenuRef.current!.contains(event.target as Node);
+      setisOpen(stillFocused);
+    };
+
+    document.addEventListener("mousedown", onClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", onClickOutside);
+    };
+  }, [listMenuRef, menuButtonRef, isOpen]);
+
+  const onClick = (v: string | T) => {
+    if (allowMultiselect) {
+      if (Array.isArray(value)) {
+        const newValue = toggleArray(v, value) as any[];
+        onChange(newValue);
+      }
+    } else {
+      onChange(v);
+    }
+    // Close the dropdown after user makes a selection only if it isn't a multiselect
+    if (!allowMultiselect) {
+      setisOpen(false);
+    }
+  };
+
+  const option =
+    optionRenderer ||
+    ((v: string | T) => (
+      <Option
+        onClick={() => onClick(v)}
+        key={`select_${v}`}
+        data-cy="searchable-dropdown-option"
+      >
+        <CheckmarkContainer>
+          {isChecked(v) && (
+            <Icon glyph="Checkmark" height={12} width={12} fill={blue.base} />
+          )}
+        </CheckmarkContainer>
+        {v}
+      </Option>
+    ));
+
+  const isChecked = (elementValue: string | T) => {
+    // If we have a search Function we can use that to determine if the option is selected
+    if (searchFunc) {
+      const isSelected =
+        (value as any).filter((o) => searchFunc(elementValue, o)).length > 0;
+      return isSelected;
+    }
+
+    if (typeof value === "string") {
+      return value === elementValue;
+    }
+    if (Array.isArray(value)) {
+      // v is included in value
+      return (value as any).filter((v) => v === elementValue).length > 0;
+    }
+  };
+
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value: searchTerm } = e.target;
+    setSearch(searchTerm);
+    let filteredOptions = [];
+    if (searchFunc) {
+      // Alias the array as any to avoid TS error https://github.com/microsoft/TypeScript/issues/36390
+      filteredOptions = (options as any).filter((o) =>
+        searchFunc(searchTerm, o)
+      );
+    } else {
+      filteredOptions = (options as any).filter(
+        (o) => o.toLowerCase().indexOf(searchTerm.toLowerCase()) !== -1
+      );
+    }
+    setVisibleOptions(filteredOptions);
+  };
+
+  return (
+    <>
+      <Label htmlFor="searchable-dropdown">{label}</Label>
+      <Wrapper>
+        <StyledButton
+          ref={menuButtonRef} // @ts-expect-error
+          onClick={() => setisOpen((curr) => !curr)}
+          data-cy="searchable-dropdown"
+          id="searchable-dropdown"
+          value={value}
+        >
+          <ButtonContent>
+            <LabelWrapper>
+              <Body data-cy="dropdown-value">{value}</Body>
```

Comment:
When multi-select is used and `value` is an array, it'd be nice to display the values here as a comma-separated list rather than just concatenating them
<img width="140" alt="image" src="https://user-images.githubusercontent.com/9298431/129906475-3a3ad682-543f-45d3-a0a8-c474cd40ee30.png">

Reply:
done